system/security/LdapSecurity/ldapsecurity.cpp

```diff
@@ -1014,6 +1016,22 @@ SecAccessFlags CLdapSecManager::authorizeFileScope(ISecUser & user, const char *
     if(filescope == 0 || filescope[0] == '\0')
         return SecAccess_Full;
 
+    //Preprocess "HpccInternal::" scopes, since they are not managed by LDAP
+    //Grant user access to their own hpccinternal::<user> scope, deny if anything else
+    if(0 == strnicmp(m_hpccInternalScope.str(), filescope, m_hpccInternalScope.length()))
+    {
+        StringBuffer userName;
+        for (const char * p = &filescope[m_hpccInternalScope.length()]; *p && *p != ':'; p++)//extract scope username
+            userName.append(*p);
+        if(userName.length() == strlen(user.getName()) &&
```

Comment:
minor: there's no point in the length check strieq will check length equality anyway (via stricmp)